src/analyses/variable-sensitivity/full_array_abstract_object.cpp

```diff
@@ -376,13 +376,10 @@ abstract_object_pointert full_array_abstract_objectt::write_leaf_element(
     }
     else
     {
-      auto const old_value = result->map.find(index_value);
+      auto old_value = result->map.find(index_value);
       if(old_value.has_value())
```

Comment:
It may not change the value that is stored but breaking sharing is a serious concern.  Please check that the sharing map does actually handle that case correctly.